lightning/src/ln/channel.rs

```diff
@@ -444,6 +444,15 @@ pub(super) struct Channel<Signer: Sign> {
 	///
 	/// See-also <https://github.com/lightningnetwork/lnd/issues/4006>
 	pub workaround_lnd_bug_4006: Option<msgs::FundingLocked>,
+
+	#[cfg(any(test, feature = "fuzztarget"))]
+	// When receive an HTLC fulfill on an outbound path, we may immediately fulfill the
+	// corresponding HTLC on the inbound path. If, then, the outbound path channel is
+	// disconnected and reconnected, they may re-broadcast their update_fulfill_htlc,
+	// causing a double-claim. This is fine, but as a sanity check in our failure to
```

Comment:
nit: I prefer the commit term "duplicate" and precise it's a lawful, correct behavior? "Double-claim" would be more a bug where we increase twice counterparty balance for the same `update_fulfill_htlc`